src/Compilers/CSharp/Portable/Binder/Binder_Conversions.cs

```diff
@@ -186,8 +178,8 @@ private BoundExpression CreateImplicitNewConversion(SyntaxNode syntax, BoundExpr
                             diagnostics);
                         break;
 
-                    default:
-                        throw ExceptionUtilities.UnexpectedValue(type.Kind);
+                    case var value:
+                        throw ExceptionUtilities.UnexpectedValue(value);
```

Comment:
💡 I think we should do this everywhere. it just pass the already created temp for switch, and you don't have to keep them insync in the source.